src/test/java/com/wikia/webdriver/common/core/api/GraphApi.java

```diff
@@ -0,0 +1,113 @@
+package com.wikia.webdriver.common.core.api;
+
+import com.wikia.webdriver.common.core.XMLReader;
+import com.wikia.webdriver.common.logging.PageObjectLogging;
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import org.apache.commons.lang.exception.ExceptionUtils;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.entity.UrlEncodedFormEntity;
+import org.apache.http.client.methods.HttpDelete;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.client.utils.URIBuilder;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.message.BasicNameValuePair;
+import org.apache.http.util.EntityUtils;
+import org.openqa.selenium.WebDriverException;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.HashMap;
+
+public class GraphApi {
+
+    private static final String ERROR_MESSAGE = "Problem with Graph API call used to create new facebook test user";
+    private static final String URI_SYNTAX_EXCEPTION = "URI_SYNTAX EXCEPTION";
+    private static ArrayList<BasicNameValuePair> PARAMS;
+    private static final String wikia_production_app_access_token = XMLReader
+        .getValue("ci.user.facebook.prod.accessToken");
+
+    public HashMap<String, String> createFacebookTestUser(String appId) {
+        try {
+            HttpResponse response = createTestUser(appId);
+            String entity = EntityUtils.toString(response.getEntity());
+            return new Gson().fromJson(entity, new TypeToken<HashMap<String, String>>(){}.getType());
+        } catch (IOException e) {
+            PageObjectLogging.log(URI_SYNTAX_EXCEPTION, ExceptionUtils.getStackTrace(e), false);
+            throw new WebDriverException(ERROR_MESSAGE);
+        } catch (URISyntaxException e) {
+            PageObjectLogging.log(URI_SYNTAX_EXCEPTION, ExceptionUtils.getStackTrace(e), false);
+            throw new WebDriverException(ERROR_MESSAGE);
+        }
+
+    }
+
+    public HashMap<String, String> deleteFacebookTestUser(String userId) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The return type of this method should be an interface such as "Map" rather than the implementation "HashMap". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-s1/coding_rules#rule_key=squid%3AS1319)